controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -130,25 +139,43 @@ func (r *StorageConsumerReconciler) initReconciler(request reconcile.Request) {
 	r.cephFilesystemSubVolumeGroup.Namespace = r.namespace
 
 	r.cephClientRBDProvisioner = &rookCephv1.CephClient{}
-	r.cephClientRBDProvisioner.Name = fmt.Sprintf("%s-%s", "cephclient-rbd-provisioner", r.storageConsumer.Name)
+	cephClientName := &cephClientFormat{StorageConsumerName: r.storageConsumer.Name, ClaimID: "rbd", CephUserType: "provisioner"}
+	r.cephClientRBDProvisioner.Name = generateHashForCephClient(cephClientName)
```

Comment:
Using a struct to serialize is an implementation detail and as such, I would expect it to be hidden inside ` generateHashForCephClient`.
```suggestion
	r.cephClientRBDProvisioner.Name = generateHashForCephClient(r.storageConsumer.Name, "rbd", "provisioner")
```